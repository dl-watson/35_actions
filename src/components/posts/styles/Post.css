* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

.Post {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    text-align: left;
    align-items: center;


    section {
        display: flex;
        flex-direction: column;
    }

    p {
        text-indent: 10px;
    }

    img {
        width: 15px;
        height: 15px;
        margin: 5px;
        cursor: pointer;
    }

    img:hover {
        transform: scale(1.2);
    }
}