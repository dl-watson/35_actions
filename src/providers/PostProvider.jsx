import React, { createContext, useContext, useReducer } from "react";
import PropTypes from "prop-types";
import { postReducer, initialState } from "../reducers/postReducer";

const PostContext = createContext(null);

export const PostProvider = ({ children }) => {
  const [state, dispatch] = useReducer(postReducer, initialState);

  return (
    <PostContext.Provider value={{ state, dispatch }}>
      {children}
    </PostContext.Provider>
  );
};

export const useSelector = (selector) => {
  const { state } = useContext(PostContext);

  return selector(state);
};

export const useDispatch = () => {
  const { dispatch } = useContext(PostContext);
  return dispatch;
};

PostProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
