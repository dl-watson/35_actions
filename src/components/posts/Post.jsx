import React from "react";
import PropTypes from "prop-types";
import { deletePost } from "../../actions/postActions";
import styles from "./styles/Post.css";
import deleteIcon from "/public/img/delete.png";
import { getComments } from "../../selectors/commentSelectors";
import { useDispatch, useSelector } from "react-redux";

const Post = ({ id, title, body }) => {
  const dispatch = useDispatch();
  const { comments } = useSelector(getComments);
  console.log(comments);

  const handleClick = () => {
    dispatch(deletePost(id));
  };

  return (
    <div className={styles.Post}>
      <img src={deleteIcon} onClick={handleClick} data-testid="delete-icon" />
      <section>
        <h2 data-testid="title">{title}</h2>
        <p>{body}</p>
        <div>{comments}</div>
      </section>
    </div>
  );
};

Post.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
};

export default Post;
