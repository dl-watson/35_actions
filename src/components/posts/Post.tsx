import React from "react";
import { deletePost } from "../../actions/postActions";
import * as styles from "./styles/Post.css";
const deleteIcon = require("/public/img/delete.png");
import { useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { deletePostComments } from "../../actions/commentActions";

type PostProps = {
  id: number, 
  title: string, 
  body: string,
  postIndex: number 
}

const Post = (props: PostProps) => {
  const dispatch = useDispatch();

  const handleDelete = () => {
    dispatch(deletePost(props.id));
    dispatch(deletePostComments(props.postIndex));
  };

  return (
    <div className={styles.Post}>
      <img src={deleteIcon} onClick={handleDelete} data-testid="delete-icon" />
      <Link to={`/${props.postIndex}`}>
        <section>
          <h2 data-testid="title">{props.title}</h2>
          <p>{props.body}</p>
        </section>
      </Link>
    </div>
  );
};

export default Post;
