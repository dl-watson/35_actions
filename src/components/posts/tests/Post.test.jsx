import React from "react";
import { render, cleanup } from "@testing-library/react";
import Post from "../Post";
import { PostProvider } from "../../../providers/PostProvider";

describe("Post component", () => {
  afterEach(() => cleanup());
  it("renders Post", () => {
    const { asFragment } = render(
      <PostProvider>
        <Post posts={{ id: Date.now(), title: "", body: "" }} />
      </PostProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
