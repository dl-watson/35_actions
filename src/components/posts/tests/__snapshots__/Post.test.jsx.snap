// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component renders Post 1`] = `
<DocumentFragment>
  <div
    class="Post"
  >
    <img
      data-testid="delete-icon"
      src="/path/image"
    />
    <section>
      <h2
        data-testid="title"
      />
      <p />
    </section>
  </div>
</DocumentFragment>
`;
